luaparse.js

```diff
@@ -907,15 +912,27 @@
           // Return it as is, without translating the byte.
           return '\\' + input.slice(sequenceStart, index);
         }
+        return raise({}, errors.hexadecimalDigitExpected, '\\' + input.slice(sequenceStart, index + 2));
+      case '\\':
+      case '\'':
+      case '"':
+      case '[':
+      case ']':
         return '\\' + input.charAt(index++);
       default:
         // \ddd, where ddd is a sequence of up to three decimal digits.
-        if (isDecDigit(input.charCodeAt(index))) {
-          while (isDecDigit(input.charCodeAt(++index)));
-          return '\\' + input.slice(sequenceStart, index);
+        while (isDecDigit(input.charCodeAt(index)) && index - sequenceStart < 3) index++;
+
+        if (index === sequenceStart && options.luaVersion >= 5.2) {
```

Comment:
I changed it to string.

` >= ` Seams to be fair for now as most update will still probably persist for a reasonable amount of future versions. And if a new version break it, changes will have to be made anyway.

Cheers